main.go

```diff
@@ -73,85 +73,94 @@ func main() {
 		case "phpapi":
-			secure, err := util.ParseSecure(ss.Token)
+			secure, err := util.ParseSecure(submSpec.Token)
 			if err != nil {
 				klog.Fatalf("Error parsing secure token: %v", err)
 			}
-			ds = phpapi.NewPHPAPI(secure.ApiKey)
+			datastore = phpapi.NewPHPAPI(secure.ApiKey)
 		case "k8s":
-			ds = subk8s.NewK8sDatastore(ss.ClusterId, stopCh)
+			datastore = subk8s.NewK8sDatastore(submSpec.ClusterId, stopCh)
 		default:
-			panic("No backend was specified")
+			klog.Fatalf("Invalid backend '%s' was specified", submSpec.Broker)
 		}
+
 		klog.V(6).Infof("Creating new datastore syncer")
-		dsSyncer := datastoresyncer.NewDatastoreSyncer(ss.ClusterId, ss.Namespace, kubeClient, submarinerClient, submarinerInformerFactory.Submariner().V1().Clusters(), submarinerInformerFactory.Submariner().V1().Endpoints(), ds, ss.ColorCodes, localCluster, localEndpoint)
+		dsSyncer := datastoresyncer.NewDatastoreSyncer(submSpec.ClusterId, submSpec.Namespace, kubeClient, submarinerClient,
+			submarinerInformerFactory.Submariner().V1().Clusters(), submarinerInformerFactory.Submariner().V1().Endpoints(), datastore,
+			submSpec.ColorCodes, localCluster, localEndpoint)
 
 		kubeInformerFactory.Start(stopCh)
 		submarinerInformerFactory.Start(stopCh)
 
 		klog.V(4).Infof("Starting controllers")
+
 		var wg sync.WaitGroup
 		wg.Add(3)
-
 		go func() {
 			defer wg.Done()
-			ce.StartEngine(true)
+			if err = cableEngine.StartEngine(true); err != nil {
+				klog.Fatalf("Error starting the cable engine: %v", err)
+			}
 		}()
 
 		go func() {
 			defer wg.Done()
 			if err = tunnelController.Run(stopCh); err != nil {
-				klog.Fatalf("Error running tunnel controller: %s", err.Error())
+				klog.Fatalf("Error running tunnel controller: %v", err)
 			}
 		}()
 
 		go func() {
 			defer wg.Done()
 			if err = dsSyncer.Run(stopCh); err != nil {
-				klog.Fatalf("Error running datastoresyncer controller: %s", err.Error())
+				klog.Fatalf("Error running datastoresyncer controller: %v", err)
 			}
 		}()
 
 		wg.Wait()
 	}
 
 	leClient, err := kubernetes.NewForConfig(rest.AddUserAgent(cfg, "leader-election"))
-	eventBroadcaster := record.NewBroadcaster()
-	eventBroadcaster.StartLogging(klog.V(4).Infof)
-	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "submariner-controller"})
 	if err != nil {
 		klog.Fatal(err)
 	}
+
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(klog.V(4).Infof)
+	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "submariner-controller"})
+
 	startLeaderElection(leClient, recorder, start)
 	klog.Fatal("All controllers stopped or exited. Stopping main loop")
 }
 
 func startLeaderElection(leaderElectionClient kubernetes.Interface, recorder record.EventRecorder, run func(ctx context.Context)) {
 	id, err := os.Hostname()
 	if err != nil {
-		klog.Fatalf("error getting hostname: %s", err.Error())
+		klog.Fatalf("error getting hostname: %v", err)
```

Comment:
The initial version seemed fine, why the change here?

Reply:
I'm not exactly sure what's better - still learning Go. FWIU, Error() returns the string representation, %v prints the default format for whatever type it is. Elsewhere in the code base, it uses %v for error so I changed this too be consistent.

Reply:
ack :)